test/kind_test_utils.sh

```diff
@@ -0,0 +1,645 @@
+# docker image to GCR, deploy it on a cluster, and run tests against
+# that deployment.
+#
+
+set -euo pipefail
+
+ROOT=`realpath $(dirname $0)/..`
+
+travis_fold_start() {
+    echo -e "travis_fold:start:$1\r\033[33;1m$2\033[0m"
+}
+
+travis_fold_end() {
+    echo -e "travis_fold:end:$1\r"
+}
+
+helm_setup() {
+    helm init
+    # wait for tiller ready
+    while true; do
+      kubectl --namespace kube-system get pod|grep tiller|grep Running
+      if [[ $? -eq 0 ]]; then
+          break
+      fi
+      sleep 1
+    done
+}
+export -f helm_setup
+
+#gcloud_login() {
+#    KEY=${HOME}/gcloud-service-key.json
+#    if [ ! -f $KEY ]
+#    then
+#	echo $FISSION_CI_SERVICE_ACCOUNT | base64 -d - > $KEY
+#    fi
+#
+#    gcloud auth activate-service-account --key-file $KEY
+#}
+
+getVersion() {
+    echo $(git rev-parse HEAD)
+}
+
+getDate() {
+    echo $(date -u +'%Y-%m-%dT%H:%M:%SZ')
+}
+
+getGitCommit() {
+    echo $(git rev-parse HEAD)
+}
+
+setupCIBuildEnv() {
+    export REPO=localhost:5000
+#    export REPO=gcr.io	
+#    export REPO=gcr.io/$GKE_PROJECT_NAME
+    export IMAGE=fission-bundle
+    export FETCHER_IMAGE=$REPO/fetcher
+    export BUILDER_IMAGE=$REPO/builder
+#    export TAG=test-${TRAVIS_BUILD_ID}
+    export TAG=test
+    export PRUNE_INTERVAL=1 # this variable controls the interval to run archivePruner. The unit is in minutes.
+    export ROUTER_SERVICE_TYPE=LoadBalancer
+    export SERVICE_TYPE=LoadBalancer
+    export PRE_UPGRADE_CHECK_IMAGE=$REPO/pre-upgrade-checks
+}
+
+setupIngressController() {
+    # set up NGINX ingress controller
+#    kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin --user $(gcloud config get-value account) || true
+    kubectl create clusterrolebinding cluster-admin-binding --clusterrole cluster-admin  || true
+    kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.25.1/deploy/static/mandatory.yaml || true
+    kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.25.1/deploy/static/provider/cloud-generic.yaml || true
+}
+
+removeIngressController() {
+    # set up NGINX ingress controller
+    kubectl delete clusterrolebinding cluster-admin-binding || true
+    kubectl delete -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.25.1/deploy/static/provider/cloud-generic.yaml || true
+    kubectl delete -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.25.1/deploy/static/mandatory.yaml || true
+}
+
+build_and_push_go_mod_cache_image() {
+    image_tag=$1
+    travis_fold_start go_mod_cache_image $image_tag
+
+#    gcloud_login
+
+    if ! gcloud docker -- pull $image_tag >/dev/null 2>&1 ; then
+      docker build -q -t $image_tag -f $ROOT/cmd/fission-bundle/Dockerfile.fission-bundle --target godep --build-arg GITCOMMIT=$(getGitCommit) --build-arg BUILDDATE=$(getDate) --build-arg BUILDVERSION=$(getVersion) .
```

Comment:
Is this still required with Skaffold in place?

Reply:
@vishal-biyani Changed this because it was giving error for pip3 command not found
```/bin/sh: pip3: not found
The command '/bin/sh -c apk add --no-cache python3 python3-dev build-base libev-dev libffi-dev &&     pip3 install --upgrade pip &&     rm -r /root/.cache' returned a non-zero code: 127
##[error]Process completed with exit code 127.```